src/main/java/org/apache/datasketches/kll/KllDirectCompactItemsSketch.java

```diff
@@ -0,0 +1,307 @@
+    return serDe.deserializeFromMemory(mem, offset, 2)[1];
+  }
+
+  @Override
+  public T getMinItem() {
+    if (sketchStructure == COMPACT_EMPTY || isEmpty()) {
+      throw new SketchesArgumentException(EMPTY_MSG);
+    }
+    if (sketchStructure == COMPACT_SINGLE) {
+      return serDe.deserializeFromMemory(mem, DATA_START_ADR_SINGLE_ITEM, 1)[0];
+    }
+    //sketchStructure == COMPACT_FULL
+    final int offset = DATA_START_ADR + getNumLevels() * Integer.BYTES;
+    return serDe.deserializeFromMemory(mem, offset, 1)[0];
+  }
+
+  @Override
+  public long getN() {
+    if (sketchStructure == COMPACT_EMPTY) { return 0; }
+    if (sketchStructure == COMPACT_SINGLE) { return 1; }
+    return getMemoryN(mem);
+  }
+
+  //restricted
+
+  private int getCompactDataOffset() { //Sketch cannot be empty
+    return sketchStructure == COMPACT_SINGLE
+        ? DATA_START_ADR_SINGLE_ITEM
+        : DATA_START_ADR + getNumLevels() * Integer.BYTES + getMinMaxSizeBytes();
+  }
+
+  @Override
+  int getM() {
+    return getMemoryM(mem);
+  }
+
+  @Override
+  MemoryRequestServer getMemoryRequestServer() {
+    return null; // not used. Must return null.
+  }
+
+  @Override
+  int getMinK() {
+    if (sketchStructure == COMPACT_EMPTY || sketchStructure == COMPACT_SINGLE) { return getMemoryK(mem); }
+    return getMemoryMinK(mem);
+  }
+
+  @Override
+  byte[] getMinMaxByteArr() {
+    if (sketchStructure == COMPACT_EMPTY || getN() == 0) { return new byte[0]; }
+    if (sketchStructure == COMPACT_SINGLE) {
+      final int bytesSingle = serDe.sizeOf(mem, DATA_START_ADR_SINGLE_ITEM, 1);
+      final byte[] byteArr = new byte[2 * bytesSingle];
+      mem.getByteArray(DATA_START_ADR_SINGLE_ITEM, byteArr, 0, bytesSingle);
+      mem.getByteArray(DATA_START_ADR_SINGLE_ITEM, byteArr, bytesSingle, bytesSingle);
+    }
+    //sketchStructure == COMPACT_FULL
+    final int offset = DATA_START_ADR + getNumLevels() * Integer.BYTES;
+    final int bytesMinMax = serDe.sizeOf(mem, offset, 2);
+    final byte[] byteArr = new byte[bytesMinMax];
+    mem.getByteArray(offset, byteArr, 0, bytesMinMax);
+    return byteArr;
+  }
+
+  @Override
+  int getMinMaxSizeBytes() {
+    if (sketchStructure == COMPACT_EMPTY || getN() == 0) { return 0; }
+    if (sketchStructure == COMPACT_SINGLE) {
+      return serDe.sizeOf(mem, DATA_START_ADR_SINGLE_ITEM, 1) * 2;
+    }
+    //sketchStructure == COMPACT_FULL
+    final int offset = DATA_START_ADR + getNumLevels() * Integer.BYTES;
+    return serDe.sizeOf(mem, offset, 2);
+  }
+
+  @Override
+  T[] getRetainedItemsArray() {
+    final int numRet = getNumRetained();
+    final T[] outArr = (T[]) Array.newInstance(serDe.getClassOfT(), numRet);
+    if (sketchStructure == COMPACT_EMPTY) { return outArr; }
+    if (sketchStructure == COMPACT_SINGLE) {
+      outArr[0] = getSingleItem();
+      return outArr;
+    }
+    final int offset = getCompactDataOffset(); //both single & full
```

Comment:
single doesn't seem to reach this point

Reply:
This actually works because `getSingleItem(`) does the work. However, this method would be easier to read if I had removed the COMPACT_SINGLE case like I did with `getRetainedItemsByteArr()` and `getRetainedItemsSizeBytes()`, because the `getCompactDataOffset()` computes the correct offset for both Compact Single and Full cases.

Reply:
Fixed.